small-talk/db/registerToDB.js

```diff
@@ -5,18 +5,31 @@ import bcrypt from "bcryptjs";
 export default async function handleRegister(username, password, firstname, lastname, dob){ // function that sends the credentials to the database
   const mongoboi = new Mongoboi(uri, "Users");
   const date = new Date(dob).getTime() / 1000; // return the inputted date as a Unix timestamp
-  await mongoboi.connect();
-
-  const hashedpass = await bcrypt.hash(password, 10);
-
+  //await mongoboi.connect();
+  
   const newUser = { // schema for holding the values into certain fields for database organization
     username: username,
-    password: hashedpass,
     firstname: firstname,
     lastname: lastname,
     dob: date,
   };
 
+  bcrypt.genSalt(10, async function(saltError, salt){ // generate a salt for password
+    if (saltError) {
+      console.log('Error salting password', saltError)
+      return null
+    }
+    else {
+      bcrypt.hash(password, salt, function (hashError, hash){ // use hash algorithm to generate hashed password w/ salt
```

Comment:
For bcrypt.hash and bcrypt.gen since they're asynchronous I believe it may be better to use them in a try catch block while using await. This approach avoids the risk of inserting incomplete user records without properly hashed passwords due to the asynchronous nature of the bcrypt operations not being awaited.

Below is a version of the same code but with the try catch block and using await. If this for some reason produces any errors though then please disregard. 

I won't impose a request changes on this comment but I do think it should be noted that the current way could potentially cause the storage of an unsalted password.

```
try {
  const salt = await bcrypt.genSalt(10);
  const hashedPassword = await bcrypt.hash(password, salt);
  newUser.password = hashedPassword;
} catch (error) {
  console.error("Error salting or hashing password:", error);
  return null; // Proper error handling
}
```


Other than that though, looks great. I'll approve once the changes are made or there's a response to this comment mentioning why it isn't necessary or couldn't be implemented.

Reply:
Yes, that is correct, thanks for the suggestion! I initially had used awaits but I guess my VSC didn't like that haha (it said it had no effect??) I'll push those changes shortly.